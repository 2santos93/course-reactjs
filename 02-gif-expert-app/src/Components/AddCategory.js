import PropTypes from 'prop-types';
import { useState } from "react";

const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');

    const inputHandler = (e) => {
        const value = e.target.value;
        setinputValue(value);
    }

    const submitHandler = (e) => {
        e.preventDefault();
        
        if(inputValue.trim().length <= 2) return;

        setCategories( (categories) => [inputValue, ...categories]);
        setinputValue('');
    }

    return (
        <form onSubmit={submitHandler}>
            <input 
                type="text"
                value={inputValue}
                onChange={inputHandler}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};

export default AddCategory;
