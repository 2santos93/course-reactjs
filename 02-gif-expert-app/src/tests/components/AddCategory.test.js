import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import AddCategory from '../../Components/AddCategory'; 

describe('tests for AddCategory component', () => {
    
    let setCategoriesMock = jest.fn();    
    let wrapper = shallow(<AddCategory setCategories={setCategoriesMock}/>);

    beforeEach(() => {
        jest.clearAllMocks();

        setCategoriesMock = jest.fn();    
        wrapper = shallow(<AddCategory setCategories={setCategoriesMock}/>);
    });

    test('should render ok', () => {
        
        expect(wrapper).toMatchSnapshot();

    });
   
    test('should show the value in the input', ()=>{
        const inputText = "Hi im goku";
        const input = wrapper.find('input');

        input.simulate('change', {target: {value: inputText}});
        const inputAfter = wrapper.find('input');

        expect(inputAfter.prop('value')).toBe(inputText);

    });

    test('shouldn\'t execute setCategories in submit event', () => {

        const form = wrapper.find('form');

        form.simulate('submit', {preventDefault(){}});

        expect(setCategoriesMock).not.toBeCalled();
    });

    test('should execute setCategories in submit event', () => {

        const inputText = "Hi im goku";
        const input = wrapper.find('input');

        input.simulate('change', {target: {value: inputText}});

        const form = wrapper.find('form');

        form.simulate('submit', {preventDefault(){}});

        const inputAfter = wrapper.find('input');

        expect(setCategoriesMock).toBeCalledTimes(1);
        expect(inputAfter.prop('value')).toBe('');
    });

});
