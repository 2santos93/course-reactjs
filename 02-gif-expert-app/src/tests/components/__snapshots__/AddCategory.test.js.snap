// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for AddCategory component should render ok 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
